frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx

```diff
@@ -156,14 +169,44 @@ export default class NativeQueryEditor extends Component {
   componentDidMount() {
     this.loadAceEditor();
     document.addEventListener("keydown", this.handleKeyDown);
+    document.addEventListener("contextmenu", this.handleRightClick);
   }
 
+  handleRightClick = event => {
+    // For some reason the click doesn't target the selection element directly.
+    // We check if it falls in the selections bounding rectangle to know if the selected text was clicked.
+    const selection = document.querySelector(".ace_selection");
+    if (!selection) {
+      return;
+    }
+    const { clientX, clientY } = event;
```

Comment:
I like to put functions like this in `metabase/lib/dom`, maybe something like `isEventOverElement(event, element)` or something.